internal/controllers/machinedeployment/machinedeployment_sync.go

```diff
@@ -563,14 +563,6 @@ func calculateStatus(allMSs []*clusterv1.MachineSet, newMS *clusterv1.MachineSet
 	if totalReplicas-availableReplicas < 0 {
 		status.Phase = string(clusterv1.MachineDeploymentPhaseScalingDown)
 	}
```

Comment:
Looking at the 3 if's above. Could it be that there are cases where we are not setting phase?

Reply:
totalReplicas-availableReplicas is a little bit strange, but I don't have historical context
What I did is to make sure we set to unknown as a fallback.

Reply:
Sounds fine. I have a relatd finding on the counter PR, but let's handle that one there